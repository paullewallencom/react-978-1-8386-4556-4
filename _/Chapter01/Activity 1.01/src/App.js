import React, { Component } from "react";

const Header = props => <h1>{props.title}</h1>;

const InventoryItem = props => (
  <div className="InventoryItem">
    <strong>{props.itemName}</strong>
    <hr />
    <p>{props.itemPrice}</p>
  </div>
);

class App extends Component {
  constructor(props) {
    super(props);

    this.state = {
      items: [
        { itemName: "Shoe", itemPrice: 5 },
        { itemName: "Sock", itemPrice: 3 }
      ]
    };
  }
  render() {
    return (
      <div className="App">
        <Header title="BuyStuff" />
        <InventoryItem
          itemName={this.state.items[0].itemName}
          itemPrice={this.state.items[0].itemPrice}
        />
        <InventoryItem
          itemName={this.state.items[1].itemName}
          itemPrice={this.state.items[1].itemPrice}
        />
      </div>
    );
  }
}

export default App;
