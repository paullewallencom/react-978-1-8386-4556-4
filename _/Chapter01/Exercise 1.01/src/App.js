import React from "react";

class App extends React.Component {
  render() {
    return (
      <div className="Example" id="my-element">
        <div
          className="greeting"
          style={{ background: "black", color: "white" }}
          onClick={() => alert("hello")}
        >
          Hello World
        </div>
      </div>
    );
  }
}

export default App;
