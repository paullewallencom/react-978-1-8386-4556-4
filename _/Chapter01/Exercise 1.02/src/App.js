import React from "react";

const App = props => (
  <div>
    <h1>My App</h1>
    <br />
    Hello React!
  </div>
);

export default App;
