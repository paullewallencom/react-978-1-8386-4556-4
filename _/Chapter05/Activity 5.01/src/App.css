.container {
  max-width: 960px;
  margin: 0 auto;
  padding: 0 10%;
}

.center {
  text-align: center;
}
