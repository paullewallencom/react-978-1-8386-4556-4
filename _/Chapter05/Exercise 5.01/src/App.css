.App {
  text-align: left;
  margin: 50px;
}
