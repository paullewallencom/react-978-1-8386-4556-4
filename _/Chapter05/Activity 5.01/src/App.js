import React from "react";
// styles
import "./App.css";
// components
import CommentSection from "./components/CommentSection/";
// data
import { comments } from "./comments.json";

function App() {
  return (
    <div className="container">
      <CommentSection comments={comments} />
    </div>
  );
}

export default App;
