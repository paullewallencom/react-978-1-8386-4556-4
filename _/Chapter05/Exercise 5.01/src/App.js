import React from "react";
import "./App.css";
import Profile from "./Profile";

const user = {
  name: "Brian",
  interests: "Reading, Swimming, Technology",
  age: 9,
  image:
    "https://images.unsplash.com/photo-1470071131384-001b85755536?auto=format&fit=crop&w=200&q=80",
  color: "Red",
  movie: "Star Wars"
};

function App() {
  return (
    <div className="App">
      <Profile user={user} />
    </div>
  );
}

export default App;
