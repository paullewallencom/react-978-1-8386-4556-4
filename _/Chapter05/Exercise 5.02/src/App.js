import React from "react";
import "./App.css";
import Loader from "./Loader";

function App() {
  return (
    <div className="App">
      <Loader spinnerSize={60} isVisible={true} />,
    </div>
  );
}

export default App;
