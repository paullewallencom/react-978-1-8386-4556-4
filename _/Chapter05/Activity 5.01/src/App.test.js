import React from "react";
import { render, cleanup } from "@testing-library/react";
import App from "./App";

afterEach(cleanup);

it("renders", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

test("renders div with class container", () => {
  const { container } = render(<App />);
  expect(container.firstChild.classList.contains("container")).toBe(true);
});

test("renders component with class comments", () => {
  const { container } = render(<App />);
  expect(container.firstChild.firstChild.classList.contains("comments")).toBe(
    true
  );
});
