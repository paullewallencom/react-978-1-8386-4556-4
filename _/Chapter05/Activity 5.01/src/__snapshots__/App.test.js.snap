// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <section
      class="comments"
    >
      <h1>
        Comments
      </h1>
      <div
        class="comment"
      >
        <div
          class="comment_container"
        >
          <img
            alt="John Smith"
            class="comment_profile"
            src="https://images.unsplash.com/photo-1522075469751-3a6694fb2f61?auto=format&fit=crop&w=200&q=80"
          />
          <div
            class="comment_content"
          >
            <h2
              class="comment_name"
            >
              John Smith
            </h2>
            <strong
              class="comment_time"
            >
              Oct 02, 2019
            </strong>
            <p>
              This is an awesome page. Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean scelerisque, purus ac feugiat eleifend, ex.
            </p>
            <button
              class="comment_button"
            >
              Reply
            </button>
          </div>
        </div>
        <div
          class="comment"
        >
          <div
            class="comment_container"
          >
            <img
              alt="Karen Michaels"
              class="comment_profile"
              src="https://images.unsplash.com/photo-1567336273898-ebbf9eb3c3bf?auto=format&fit=crop&w=200&q=80"
            />
            <div
              class="comment_content"
            >
              <h2
                class="comment_name"
              >
                Karen Michaels
              </h2>
              <strong
                class="comment_time"
              >
                Oct 10, 2019
              </strong>
              <p>
                Lorem ipsum dolor sit amet, consectetur adipiscing elit. Aenean scelerisque, purus ac feugiat eleifend, ex.
              </p>
            </div>
          </div>
        </div>
        <div
          class="comment"
        >
          <div
            class="comment_container"
          >
            <img
              alt="Tim Parker"
              class="comment_profile"
              src="https://images.unsplash.com/photo-1505247964246-1f0a90443c36?auto=format&fit=crop&w=200&q=80"
            />
            <div
              class="comment_content"
            >
              <h2
                class="comment_name"
              >
                Tim Parker
              </h2>
              <strong
                class="comment_time"
              >
                Oct 05, 2019
              </strong>
              <p>
                Tullam elementum vulputate lectus sed pellentesque. Maecenas convallis orci at elit consectetur tempus ac id.
              </p>
            </div>
          </div>
        </div>
      </div>
      <div
        class="comment"
      >
        <div
          class="comment_container"
        >
          <img
            alt="Sarah James"
            class="comment_profile"
            src="https://images.unsplash.com/photo-1552607676-17f088307dce?auto=format&fit=crop&w=200&q=80"
          />
          <div
            class="comment_content"
          >
            <h2
              class="comment_name"
            >
              Sarah James
            </h2>
            <strong
              class="comment_time"
            >
              Sept 22, 2019
            </strong>
            <p>
              Nice! Nullam elementum vulputate lectus sed pellentesque. Maecenas convallis orci at elit consectetur tempus ac id.
            </p>
            <button
              class="comment_button"
            >
              Reply
            </button>
          </div>
        </div>
      </div>
    </section>
  </div>
</DocumentFragment>
`;
